### PR TITLE
Option to include/ignore files depending on NODE_ENV

### PR DESCRIPTION
**Pull request by emilhem**
This PR makes it so that you can put `dev`/`stage`/`prod` between `//` and `=` to make it so that files are only included if NODE_ENV is `dev`, `stage`, or `prod`.

```JavaScript
// Examples:
//
//     //dev= require foo
//     // dev =require foo
//     //dev =require foo
//     //development=require foo
//     //prod=require foo
//     //production=require foo
//     //stage=require foo
//     //staging=require foo
```

This PR needs some tests though.

### CODE REVIEW

**Comment by emilhem** [comment]
The failing test didn't happen on my machine. Probably Travis being slow resulting in a timeout error here: https://github.com/nodeca/mincer/blob/54c0db1324d84d578fb607355769ca64ba9e7c0e/test/examples_test.js#L22

**Comment by ixti** [comment]
I a strongly against this change it brings couple of problems:

1. it hard-codes possible env names which is bad on it's own
2. which is even worse, it makes file bundle different parts depending on NODE_ENV meas problems with caching at bare minimum

If you want to include different parts depending on NODE env you can create multiple files like:

``` javascript
// file: app.js
//= require foo


// file app.development.js
//= require app
//= require debug
```

And then use different bundle depending on NODE_ENV.

In other words - that is not a task to be solved by mincer.

**Comment by emilhem** [comment]
Hmm, but there are no way to ignore files with Mincer at the moment.

I'm using `Mincer.Environment`
```JavaScript
mincerEnv.appendPath("app/assets/js");
```

In the `js` folder there are a bunch of `js` files. How can I use mincer to bundle development files instead of production files when `NODE_ENV=dev` like in your example above?

I want to use non minified library versions when in development.

**Comment by ixti** [comment]
Mincer does not bundles all files it sees in appended paths. `appendPath` adds give path to the list of lookup paths, to explain:

``` javascript
assets.appendPath("app/assets/javascripts");
assets.appendPath("vendor/assets/javascripts");
```

Then, when you will use declaration like:

``` javascript
//= require foobar
```

Or with something like:

``` javascript
assets.findAsset("foobar");
```

Mincer will lookup and return first found:

- `app/assets/javascripts/foobar{.coffee,.ts,.js}`
- `vendor/assets/javascripts/foobar{.coffee,.ts,.js}`

Now let's talk about bundling and writing files themselves. If we are talking about using assets server of Mincer then you don't need to do anything extra - just require different assets depending on your NODE_ENV in your view files. For example:

``` pug
case NODE_ENV
  when "development"
    != javascript("foobar.development")
```

See examples `server.js` for `javascript` helper implementation.

If you are talking about manifest, then you also are in full control of what is gonna be bundled and written. Take a look at example [manifest.js](https://github.com/nodeca/mincer/blob/master/examples/manifest.js#L31-L35) for details - it's pretty self-explanatory.

**Comment by emilhem** [comment]
Hmm.
I tried to create a folder structure like:
```
dev/
    app.js
prod/
    app.js
js/
    app.js
```
The `prod` and `dev` `app.js` includes the `js/app.js` like this: 
```JavaScript
//= require debuglib
//= require ./../js/app.js
```

and the Mincer environment looks like:
```JavaScript
let environment = new Mincer.Environment();
environment.appendPath("app/assets/dev");
environment.appendPath("app/assets/js");
app.use("/assets", mincer.createServer(environment)); // express
```

But Mincer uses the `app.js` from the `js` directory. I want it to use the `app.js` from the `dev` directory.

**Comment by ixti** [comment]
So you have 3 files named the same in different lookup paths, so in case of:

``` javascript
environment.appendPath("app/assets/dev");
environment.appendPath("app/assets/js");
```

You will have 2 assets matching `app` name:

- app/assets/dev/app.js
- app/assets/js/app.js

And which one will be used depends only on the order of which path appended when. Basically you can make this work - but you will be struggling weird issues all the time. Just move your `dev` and `prod` under the `js`, e.g. having structure like this:

```
js/
  dev/
    app.js
  prod/
    app.js
  app.js
```

Then you will have `js/dev/app.js` like this:

``` javascript
//= require ./debuglib
//= require ../app
```

And your environment will look:

``` javascript
const environment = new Mincer.Environment();
environment.appendPath("app/assets/js");
app.use("/assets", mincer.createServer(environment));
```

**Comment by ixti** [comment]
Slightly different version that will look similar to hat you have done but that will work will look like this:

```
dev/
  app.js
prod/
  app.js
js/
  app.core.js
```

And your `dev/app.js` will look like:

``` javascript
//= require debuglib
//= require app.core
```

And your environment will look like:

``` javascript
const environment = new Mincer.Environment();
environment.appendPath("app/assets/js");
environment.appendPath("app/assets/dev");
app.use("/assets", mincer.createServer(environment));
```
